pkg/apic/apiservicerevision.go

```diff
@@ -298,7 +312,32 @@ func (c *ServiceClient) GetAPIRevisionByName(revisionName string) (*v1alpha1.API
 		}
 		return nil, nil
 	}
-	apiRevision := new(v1alpha1.APIServiceRevision)
-	json.Unmarshal(response.Body, apiRevision)
-	return apiRevision, nil
+	apiRevision := new(mv1a.APIServiceRevision)
+	err = json.Unmarshal(response.Body, apiRevision)
+	return apiRevision, err
+}
+
+func buildAPIServiceRevisionSpec(serviceBody *ServiceBody) mv1a.ApiServiceRevisionSpec {
+	return mv1a.ApiServiceRevisionSpec{
+		ApiService: serviceBody.serviceContext.serviceName,
+		Definition: mv1a.ApiServiceRevisionSpecDefinition{
+			Type:  getRevisionDefinitionType(*serviceBody),
+			Value: base64.StdEncoding.EncodeToString(serviceBody.SpecDefinition),
+		},
+	}
+}
+
+func getRevisionPrefix(serviceBody *ServiceBody) string {
+	if serviceBody.Stage != "" {
+		return sanitizeAPIName(fmt.Sprintf("%s-%s", serviceBody.serviceContext.serviceName, serviceBody.Stage))
+	}
+	return sanitizeAPIName(serviceBody.serviceContext.serviceName)
+}
+
+// getRevisionDefinitionType -
+func getRevisionDefinitionType(serviceBody ServiceBody) string {
```

Comment:
maybe this is better put as a function on ServiceBody?

Reply:
I'll look into it